### ISSUE TITLE
WIP: CUDA performances

### ISSUE THREAD

**Issue by albop** [EconForge/interpolation.py#6]
Currently, the experiments with CUDA yield terrible performances on AWS/g2.
It may be because, calculations are made on 64 bits by default.
To do :
- test 32 bits calculations
- test 64 bits on a card which supports it natively (e.g. gtx titan)
- try guvectorize(...target='gpu')

**Comment by tilman-g** [comment]
Dear albop, 
I started using the interpolation package and I am amazed by the speed-up of jitting. I would like to explore the GPU-side a bit more as I would be happy to speed things up a bit more and contribute code. Has there been any attempts so far to re-activate the cuda code and if so, what is the current status?

**Comment by albop** [comment]
No, I did not attempt to reactivate the code, mostly for lack of time and other distractions. One idea would be to tweak a bit the output of codegen. From this discussion, https://groups.google.com/a/continuum.io/forum/#!msg/numba-users/8Hn6GagrXXU/ivQa4JUzCgAJ , I gather there is nothing fundamentally wrong with this approach.
The other idea, would be to test, whether the current eval_linear function can be called on the gpu inside a cuda kernel. This is not totally impossible and would simplify things a lot. Can you try it ?

**Comment by tilman-g** [comment]
Okay, that makes sense. I am a new to GPU coding, but would really like to try. So I will look into it the next few weeks and see what I can do. I will focus on the eval_linear function for the moment.
